clarity_ext/domain/analyte.py

```diff
@@ -1,54 +1,54 @@
 from clarity_ext.domain.container import PlatePosition, Well, Container
+from clarity_ext.utils import get_and_apply
 
 
 class Analyte(object):
     """
-    Describes an Analyte in the Clarity LIMS system, including custom UDFs.
+    Describes an Analyte in the Clarity LIMS system.
 
-    Takes an analyte resource as input.
+    Expects certain mappings to UDFs in clarity. These are provided
+    in udf_map, so they can be overridden in different installations.
     """
 
-    def __init__(self, resource, container):
```

Comment:
My idea was not that it should be restricted to UDFs, but rather that it should contain all non-positional arguments, which happen to include UDFs too. As I see it, the domain object shouldn't necessarily know about UDFs as a Clarity concept. Only the `create_from_rest_resource` needs to know about that, that's why it's that method that has the udf_map.

So this is just a kwargs dictionary and it's a custom to use the name kwargs for that (keyword arguments) in Python. The idea is that listing all the possible positional arguments in the constructor becomes cumbersome:

https://github.com/search?l=Python&q=kwargs&type=Repositories&utf8=%E2%9C%93